src/Workspaces/Core/Portable/CodeActions/CodeActionWithOptions.cs

```diff
@@ -65,6 +66,14 @@ public abstract class CodeActionWithOptions : CodeAction
         /// <param name="cancellationToken">A cancellation token.</param>
         protected abstract Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(object options, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Override this method to compute the operations that implement this <see cref="CodeAction"/>. Prefer
+        /// overriding this method over <see cref="ComputeOperationsAsync(object, CancellationToken)"/> when computation
+        /// is long running and progress should be shown to the user.
+        /// </summary>
+        private protected virtual Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(object options, IProgress<CodeAnalysisProgress> progress, CancellationToken cancellationToken)
+            => ComputeOperationsAsync(options, cancellationToken);
```

Comment:
this is a method that would become purely protected once we agree that thsi api shape looks good.